nixos/modules/misc/nixpkgs.nix

```
@@ -345,6 +348,11 @@ in
       pkgs = finalPkgs.__splicedPackages;
     };
 
+    # pass along the pkgs.config, if pkgs is defined
+    # don't set as defaults so that it is merged with
+    # other values set by the module system
+    nixpkgs.config = configIfPkgsDefined;
```

Why not

```suggestion
    nixpkgs.config = if opt.pkgs.isDefined then cfg.pkgs.config else {};
```

directly?

---

True. Done.